src/main/java/core/basesyntax/serviceimpl/ShopServiceImpl.java

```
@@ -0,0 +1,24 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
```

Constructor does not validate its parameter. The task requires methods to validate non-null inputs ("Null parameters"). Add a null-check for `operationStrategy` and throw an informative RuntimeException (e.g., `new RuntimeException("OperationStrategy must not be null")`) to fail fast if a null is supplied. This follows the null-handling guidance in the checklist.